src/runtime.ts

```diff
@@ -0,0 +1,317 @@
+import type { Redis } from "ioredis";
+import { _Private, Publisher, Subscriber } from "./types";
+import { CreateResumableStreamContextOptions } from "./types";
+import { ResumableStreamContext } from "./types";
+import { createPublisherAdapter, createSubscriberAdapter } from "./ioredis-adapters";
+
+interface CreateResumableStreamContext {
+  keyPrefix: string;
+  waitUntil: (promise: Promise<unknown>) => void;
+  subscriber: Subscriber;
+  publisher: Publisher;
+}
+
+export function createResumableStreamContextFactory(defaults: _Private.RedisDefaults) {
+  return function createResumableStreamContext(
+    options: CreateResumableStreamContextOptions
+  ): ResumableStreamContext {
+    const ctx = {
+      keyPrefix: `${options.keyPrefix || "resumable-stream"}:rs`,
+      waitUntil: options.waitUntil,
+      subscriber: options.subscriber,
+      publisher: options.publisher,
+    } as CreateResumableStreamContext;
+    let initPromises: Promise<unknown>[] = [];
+
+    // Check if user has passed a raw ioredis instance
```

Comment:
The tests were failing because I failed to check the user had passed in a raw ioredis instance.